### PR TITLE
[Compute] BREAKING CHANGE: Remove unversioned image aliases

### LINKED ISSUE #27564: Remove unversioned image aliases from CLI
### Preconditions

- [X] No need to upgrade Python SDK or the Python SDK is ready.

### Related command

Any use of the -image parameter that takes in an unversioned linux image alias. 

### Resource Provider

Microsoft.Compute

### Description of Feature or Work Requested

This PR will remove the unversioned image aliases: https://github.com/Azure/azure-rest-api-specs/pull/26086
The CLI code needs to be updated to remove these aliases along with any related tests. 
This has been planned since this past March. 

Additionally, the messaging warning users of this breaking change should be removed from the vm and vmss creation cmdlets and any others that had this warning added. 

### Minimum API Version Required

2023-07-01

### Swagger PR link / SDK link

https://github.com/Azure/azure-rest-api-specs/pull/26086

### Request Example

N/A

### Target Date

2023-10-25

### PM Contact

saakpan@microsoft.com

### Engineer Contact

adsandor@microsoft.com

### Additional context

The proper communications for this breaking change were sent out repeatedly and many months ago.

### PR DESCRIPTION
**Pull request by ReaNAiveD**
**Related command**

All commands using image aliases

**Description**

Resolve #27564 

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/keyvault/_help.py [hunk 1]
```diff
@@ -58,7 +58,7 @@
         vm_secrets=$(az vm secret format -s "$secrets")
 
         az vm create -g group-name -n vm-name --admin-username deploy  \\
-          --image debian --secrets "$vm_secrets"
+          --image Debian:debian-10:10:latest --secrets "$vm_secrets"
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/keyvault/_help.py:61]:
```suggestion
          --image Debian11 --secrets "$vm_secrets"
```

File: src/azure-cli/azure/cli/command_modules/keyvault/_help.py [hunk 2]
```diff
@@ -166,7 +166,7 @@
...
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/keyvault/_help.py:169]:
```suggestion
          --image Debian11 --secrets "$vm_secrets"
```

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_image.py [hunk 3]
```diff
@@ -59,7 +59,7 @@ def test_when_alias_doc_is_missing(self, mock_get_active_cloud):
         cli_ctx = DummyCli()
         cli_ctx.cloud = mock_cloud
         images = load_images_from_aliases_doc(cli_ctx)
-        self.assertEqual(images[1], {'urnAlias': 'CentOS85Gen2', 'publisher': 'OpenLogic',
+        self.assertEqual(images[0], {'urnAlias': 'CentOS85Gen2', 'publisher': 'OpenLogic',
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_image.py:62]: Why do we need to adjust the index of `images` from 0 to 1?

**Review by zhoxing-ms — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Review by AllyW — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by AllyW — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**